oolab/src/main/java/agh/ics/oop/classTypeAndPositionTuple.java

```diff
@@ -0,0 +1,52 @@
+package agh.ics.oop;
+
+import java.util.Comparator;
+
+/* I had to make this record, because there was a bug causing MabBoundary not to delete an object from a TreeList.
+The comparator I used to automate sorting had either no information about oldPosition or newPosition.
+That flaw caused a situation in which there was no guarantee it would find an object to remove. */
+
+public record classTypeAndPositionTuple(Vector2d position, Class<?> classType) {
```

Comment:
nazwy rekordów z dużej litery. Nie jestem też pewien czy dobrze rozumiem problem, ale na pewno jeśli chce Pan porównywać typy obiektów to to dobry pomysł.